libexec/apply_catalog.rb

```diff
@@ -43,7 +44,8 @@
 Puppet[:default_file_terminus] = :file_server
 
 exit_code = 0
-Dir.mktmpdir do |moduledir|
+begin
+  moduledir = Dir.mktmpdir
```

Comment:
Should this be outside the begin so we get the 'proper' error when Dirmktmpdir fails?

Reply:
Well, we shouldn't try to remove it if we don't create it in the first place.